stores/amazon_monitoring.py

```diff
@@ -111,6 +98,33 @@
 amazon_config = {}
 
 
+class ItemsHandler:
+    @classmethod
+    def create_items_pool(cls, item_list):
+        cls.items = cycle(item_list)
+
+    @classmethod
+    def assign_next_item(cls):
+        return next(cls.items)
+
+class BadProxyCollector:
+    @classmethod
+    def __init__(cls):
+        if os.path.exists(BAD_PROXIES_PATH):
+            with open(BAD_PROXIES_PATH) as f:
+                cls.collection = json.load(f)
+        else:
+            cls.collection = {}
+
+    @classmethod
+    def collect(cls, status, connector):
+        if status == 503:
+            proxy = str(connector.proxy_url)
+            cls.collection.update({proxy : "null"})
+            with open(BAD_PROXIES_PATH, "w") as f:
+                json.dump(cls.collection, f, indent=4)
```

Comment:
Doing this write to the filesystem every loop seems like it will slow things down. I'd just keep the collection in memory, and then either schedule an additional periodic task every few minutes to flush it to disk (bbasically this code) or put the write to disk in the interrupt handler, or both.

Reply:
The way to put it in a separate periodic task is to write a new `async def` that does this, and then ends with `await asyncio.sleep(some_time)`, and add that to the `.gather` call that's bundling up all the tasks to run, I think it's in cli.py

Reply:
I've put a timer on it so that it only saves once every minute. May increase it to 5 minutes.